core/codegen/src/ident.rs

```diff
@@ -0,0 +1,99 @@
+//! A variant of Ident with more ergonomic handling of raw identifiers.
+
+use std::cmp::Ordering;
+
+use devise::syn;
+use self::syn::ext::IdentExt as _;
+use crate::proc_macro2::{Ident as Ident2, Span};
+use quote::ToTokens;
+
+#[repr(transparent)]
+#[derive(Clone, Debug, Eq, Ord)]
+pub struct Ident(Ident2);
+
+impl Ident {
+    pub fn is_raw(&self) -> bool {
+        self.0.to_string().starts_with("r#")
+    }
+
+    pub fn name(&self) -> String {
+        self.0.unraw().to_string()
+    }
+
+    pub fn prepend(&self, string: &str) -> Ident {
+        // TODO: want to use r#{}{}, but that fails (`"r#__rocket_param_enum"` is not a valid identifier)
+        Ident::from(syn::Ident::new(&format!("{}{}", string, self.unraw()), self.span()))
+    }
+
+    pub fn span(&self) -> Span {
```

Comment:
Shouldn't this come for free through the `Deref` impl?

Reply:
It should, but I ran into some kind of method resolution problem. I will take a closer look and document it.

Reply:
Without this `pub fn`, `.span()` resolves to a `proc_macro::Span` instead of a `proc_macro2::Span` and I can't figure out why.

Reply:
I believe it's <https://docs.rs/devise/0.2.0/devise/trait.Spanned.html#tymethod.span> that is being selected instead of the `Deref` impl. I have added a commit that removes this function and updates the call sites instead.